backend/src/main/java/com/bakdata/conquery/sql/execution/SqlExecutionService.java

```diff
@@ -123,15 +93,44 @@ private String getColumnName(ResultSet resultSet, int columnIndex) {
 
 	private SqlEntityResult getResultRow(ResultSet resultSet, List<ResultType<?>> resultTypes, int columnCount) throws SQLException {
 
-		String id = resultSet.getString(PID_COLUMN_INDEX);
-		Object[] resultRow = new Object[columnCount - 1];
+		final String id = resultSet.getString(PID_COLUMN_INDEX);
+		final Object[] resultRow = new Object[columnCount - 1];
 
 		for (int resultSetIndex = VALUES_OFFSET_INDEX; resultSetIndex <= columnCount; resultSetIndex++) {
-			int resultTypeIndex = resultSetIndex - VALUES_OFFSET_INDEX;
-			resultRow[resultTypeIndex] = resultTypes.get(resultTypeIndex).getFromResultSet(resultSet, resultSetIndex, this.resultSetProcessor);
+			final int resultTypeIndex = resultSetIndex - VALUES_OFFSET_INDEX;
+			resultRow[resultTypeIndex] = resultTypes.get(resultTypeIndex).getFromResultSet(resultSet, resultSetIndex, resultSetProcessor);
 		}
 
 		return new SqlEntityResult(id, resultRow);
 	}
 
+	public Result<?> fetch(Select<?> query) {
+		log.debug("Executing query: \n{}", query);
+		try {
+			return dslContext.fetch(query);
+		}
+		catch (DataAccessException exception) {
+			throw new ConqueryError.SqlError(exception);
+		}
+	}
+
+	/**
+	 * Executes the query and returns the results as a Stream.
+	 * <p>
+	 * Note: The returned Stream is resourceful. It must be closed by the caller, because it contains a reference to an open {@link ResultSet}
```

Comment:
> The returned Stream is resourceful.


![](https://c.tenor.com/A7NIlRIBtNoAAAAC/tenor.gif)

Reply:
@jnsrnhld

Reply:
Resourceful meint in dem Fall: Enthält eine Referenz auf ein geöffnetes `ResultSet` bzw. `PreparedStatement`. Der Maintainer von JOOQ benutzt das hier und da mal, deswegen hab ich das übernommen 😄 Hier z.B. https://blog.jooq.org/the-many-different-ways-to-fetch-data-in-jooq/

Reply:
Okay, der Kontext hat gefehlt :D